homeassistant/components/govee_light_api/coordinator.py

```diff
@@ -0,0 +1,122 @@
+"""Coordinator for Govee Local API."""
+
+from collections.abc import Callable
+from datetime import timedelta
+import logging
+
+from govee_local_api import GoveeController, GoveeDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_BIND_ADDRESS,
+    CONF_DISCOVERY_INTERVAL,
+    CONF_DISCOVERY_INTERVAL_DEFAULT,
+    CONF_LISENING_PORT,
+    CONF_MULTICAST_ADDRESS,
+    CONF_TARGET_PORT,
+)
+
+
+class GoveeLocalApiCoordinator(DataUpdateCoordinator):
+    """My custom coordinator."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        scan_interval: timedelta,
+        logger: logging.Logger,
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass=hass,
+            logger=logger,
+            name="GoveeLightLocalApi",
+            update_interval=scan_interval,
+        )
+
+        self._config_entry = config_entry
```

Comment:
after the `super().__init__()`, `self.config_entry` is set to the config entry. Please retype `config_entry` to `ConfigEntry` tho. Checkout the youtube coordinator on how to do that